tests/exec-tests.sh

```diff
@@ -33,8 +33,8 @@ RUN sed -i 's|</dev/tty||g' /var/lib/githooks/install.sh && \\
     # Change the base template so we can pass in the hook name and accept flags
     sed -i -E 's|echo.*Hook not run inside a git repository.*|CURRENT_GIT_DIR=".git"|' /var/lib/githooks/base-template.sh && \\
     sed -i -E 's|GITHOOKS_RUNNER=(.*)|GITHOOKS_RUNNER=\1; GITHOOKS_RUNNER="\${GITHOOKS_RUNNER:-/var/lib/githooks/base-template.sh}"|' /var/lib/githooks/base-template-wrapper.sh && \\
-    sed -i -E 's|GITHOOKS_HOOK_FOLDER=(.*)\\\|GITHOOKS_HOOK_FOLDER=\1 GITHOOKS_HOOK_FOLDER="\${HOOK_FOLDER:-\$GITHOOKS_HOOK_FOLDER}" \\\|' /var/lib/githooks/base-template-wrapper.sh && \\
-    sed -i -E 's|GITHOOKS_HOOK_NAME=(.*)\\\|GITHOOKS_HOOK_NAME=\1 GITHOOKS_HOOK_NAME="\${HOOK_NAME:-\$GITHOOKS_HOOK_NAME}" \\\|' /var/lib/githooks/base-template-wrapper.sh && \\
+    sed -i -E 's|HOOK_FOLDER=(.*)|HOOK_FOLDER="\${HOOK_FOLDER:-\1}"|' /var/lib/githooks/base-template.sh && \\
+    sed -i -E 's|HOOK_NAME=(.*)|HOOK_NAME="\${HOOK_NAME:-\1}"|' /var/lib/githooks/base-template.sh && \\
```

Comment:
Why modify these `HOOK_FOLDER` and `HOOK_NAME` lines at all? Why not just modfiy the tests steps to pass this stuff in the normal way? e.g.

```diff
- HOOK_NAME=pre-commit HOOK_FOLDER=$(pwd)/.git/hooks \
-        sh /var/lib/githooks/base-template.sh
+        sh /var/lib/githooks/base-template.sh "$(pwd)/.git/hooks/pre-commit"
```

Reply:
Correct, but this is for a later PR, so far we just modify these as before, to *directly* execute the template...
Its lots of regex replace...

Reply:
Hm, wait. It seems that (after this PR) most of the testcases actually seem to call the *wrapper*, rather than the template directly: https://github.com/rycus86/githooks/pull/111/files#diff-07d8a301a5f5b558884d60a40b4c9040

Why not revert that change and apply my above change instead? Just keep calling the script base-template directly, but pass it the right `$0` value? Not sure if that's for another PR, since you're touching this stuff already anyway, and updating the tests to the new way of passing `$0` introduced with this PR seems appropriate to me?

Reply:
Jeah, should be appropriate, but lots of work.
I decided to call the wrapper, because it tests the whole chain, and I kind of dont like this "direct" executing of the base-template, and tought, -> call the chain which is also called in the setup nothing else, thats why...

Reply:
I guess that makes sense. But if you really want to test the entire chain, shouldn't you just install the wrapper into the `.git/hooks` directory and just call that?

Alternatively, it seems you can also call `sh` in a way where it executes one script, but sets `$0` to something else. It's a bit finnicky, but can work like this:
```
matthijs@grubby:~$ cat test.sh 
#!/bin/sh
echo $0
echo $1
matthijs@grubby:~$ sh -c '. ./test.sh' /tmp/.git/hooks/pre-commit some_argument
/tmp/.git/hooks/pre-commit
some_argument
```

So the `-c` argument is a commandline to execute, the rest of the arguments is `$0`, `$1`, etc. For `-c`, I used `.` to source the script, rather than execute it, since the latter would spawn a new shell that would change `$0` to `./test.sh` again.